### PR TITLE
[GraphBolt] Add `full_nodes_itemset` function to execute `model.inference`

### PR DESCRIPTION
**Pull request by keli-wen**
## Description

Add the `full_nodes_itemset` function to obtain the complete itemset of seed nodes from the input graph.

This function is primairly designed to support the GraphBolt version `model.inference` method.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [x] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/utils/sample_utils.py [hunk 1]
```diff
@@ -6,9 +6,31 @@
 import torch
 
 from ..base import etype_str_to_tuple
+from ..itemset import ItemSet, ItemSetDict
 from ..minibatch import MiniBatch
 
 
+def full_nodes_itemset(
+    graph, name="full_seed_nodes"
```
Comment (frozenbugs) [inline python/dgl/graphbolt/utils/sample_utils.py:14]: What's the type of the graph?

File: python/dgl/graphbolt/utils/sample_utils.py [hunk 2]
```diff
@@ -6,9 +6,31 @@
 import torch
 
 from ..base import etype_str_to_tuple
+from ..itemset import ItemSet, ItemSetDict
 from ..minibatch import MiniBatch
 
 
+def full_nodes_itemset(
```
Comment (frozenbugs) [inline python/dgl/graphbolt/utils/sample_utils.py:13]: What about support range here https://github.com/dmlc/dgl/blob/c298223f5de09e9ee265cf6a9fb9145b692b4c5b/python/dgl/graphbolt/itemset.py#L66?

And then provide a util function to handle dict?

File: python/dgl/graphbolt/utils/sample_utils.py [hunk 3]
```diff
@@ -6,9 +6,31 @@
...
+    graph, name="full_seed_nodes"
+) -> Union[ItemSet, ItemSetDict]:
+    """To get the full seed nodes itemset of the graph."""
+    node_type_offset = graph.node_type_offset
```
Comment (frozenbugs) [inline python/dgl/graphbolt/utils/sample_utils.py:17]: We should avoid using csc_sampling_graph concept here.

**Review by Skeleton003 — COMMENTED** [review]
Shall we close this PR?

**Comment by keli-wen** [comment]
Done
